### DISCUSSION TITLE
Permissions Question - open wiki with private areas

### DISCUSSION THREAD

**Discussion by yoitguy** [requarks/wiki#6559] [Help / Questions] [answered]
Hi. I’m several weeks into discovering Wiki.JS. I’ve read the Users, Groups, & Permissions documentation ad nauseum, combed through permissions-related questions on github, and done a bit of experimentation. I’m clear on the difference between global and page permissions. I am wondering if I am correctly understanding what my options are in my use case.

Let’s say I want to set the wiki up for four teams with sections as follows:

- /team1/publicstuff
- /team1/privatestuff
- /team2/publicstuff
- /team3/publicstuff

I set up groups Team1, Team2, Team3.

I want everything on the wiki to be readable by everyone (including not logged in), except for anything under the privatestuff path.

I can think of two ways to accomplish this. Is there a better way?

Option 1

a.	Guest Group
i.	Add all users to the Guests group
ii.	Remove the default ALLOW page rule from /
iii.	Add ALLOW page rules for /team1/publicstuff, /team2/publicstuff, /team3/publicstuff

b. Team1 Group
i.	Add an ALLOW page rule to /team1/privatestuff

Possible Con: Since I’d like the wiki to be mostly open by default, if someone adds a path outside of /team1, /team2, /team3, then we’ll have to remember to update the Guest group to include the new path

Option 2
a.	Guest Group – Leave at defaults
b.	Team1 Group
i.	Add page rule ALLOW to /
c.	Team2 and Team3 Groups
i.	Add page rule ALLOW to /
ii.	Add page rule DENY to /team1/privatestuff

Con: If a new team is onboarded and someone forgets to add the privatestuff DENY rule to the new team group, then privatestuff will be publicly accessible

It seems like Option 1 is the better path. Looking for a sanity check of whether I am understanding this correctly and also whether there is a 3rd, easier option.

Thank you in advance!

**Comment by NGPixel** [comment, marked as answer]
You should NOT add users to the Guests group. The guests group represents users that are not logged in. Adding normal users to it would only create confusion.

The proper way to do it:

- **Guests** group
  - ALLOW path from /
  - DENY path from /team1/privatestuff

- **Users** group (new group, which contains users from all teams)
  - ALLOW path from /
  - DENY path from /team1/privatestuf (notice the missing letter at the end, see why below)

- **Team 1** Group
  - ALLOW path from /team1/privatestuff

You're running into one of the limitation of the current version: cases where you want ALLOW to override DENY. The workaround is to make the DENY path less specific than the ALLOW path. The upcoming v3 release will have a new FORCE ALLOW state that will override a DENY rule of the same specificity.

**Reply by yoitguy** [reply]
This worked like a charm and makes it much clearer, thank you Nicolas!

**Reply by pavelcdn** [reply]
Can you please tell me the approximate time frame for version 3?

**Reply by yoitguy** [reply]
https://docs.requarks.io/releases/eta

**Reply by pavelcdn** [reply]
helpful, thank you
